### PR TITLE
refactor: dataclass conversion of UserTokenBlackList

### LINKED ISSUE #6465: Refactoring model definitions to data classes
**Feature Description**
A ton of boilerplate code is written everywhere because of the redundant `__init__`, `__repr__` and other methods. Moving to the dataclass architecture would prove to get rid of this and focus mainly on data.

**Describe the solution you'd like**
Implementation of Data Classes

### PR DESCRIPTION
**Pull request by codedsun**
Fix for #6465 :refactor: dataclass conversion of UserPermission, UserTokenBlackList

#### Short description of what this resolves:
refactor: dataclass conversion of UserPermission, UserTokenBlackListT…

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by codedsun** [comment]
@iamareebjamal Review

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4820342&bid=16025263)

File: app/models/user_permission.py [hunk 1]
```diff
@@ -1,30 +1,19 @@
...
-                 anonymous_user=False, deleted_at=None):
-        self.name = name
-        self.description = description
-        self.unverified_user = unverified_user
-        self.anonymous_user = anonymous_user
-        self.deleted_at = deleted_at
-
-    def __repr__(self):
-        return '<UserPerm %r>' % self.name
+    id: int = db.Column(db.Integer, primary_key=True)
```
Comment (niranjan94) [inline app/models/user_permission.py]: Codacy Issue found: [invalid syntax](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4820342)

File: app/models/user_token_blacklist.py [hunk 2]
```diff
@@ -1,23 +1,18 @@
...
-    user = db.relationship("User", backref="token_blacklist_times", foreign_keys=[user_id])
-
-    def __init__(self, user_id=None, created_at=None, blacklisted_at=None):
-        self.user_id = user_id
-        self.created_at = created_at
-        self.blacklisted_at = blacklisted_at
-
-    def __str__(self):
-        return '<TokenBlackListTime User %s blacklisted at %s>' % (self.user, self.blacklisted_at)
+    id: int = db.Column(db.Integer, primary_key=True)
```
Comment (niranjan94) [inline app/models/user_token_blacklist.py]: Codacy Issue found: [invalid syntax](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4820342)

File: app/models/user_permission.py [hunk 3]
```diff
@@ -1,30 +1,19 @@
...
 
-    id = db.Column(db.Integer, primary_key=True)
-    name = db.Column(db.String, unique=True)
-    description = db.Column(db.String)
-
-    unverified_user = db.Column(db.Boolean)
-    anonymous_user = db.Column(db.Boolean)
-
-    def __init__(self, name, description, unverified_user=False,
-                 anonymous_user=False, deleted_at=None):
```
Comment (iamareebjamal) [inline app/models/user_permission.py]: No replacement for this logic

Comment (codedsun) [reply app/models/user_permission.py]: Shall I revert this change then?

Comment (codedsun) [reply app/models/user_permission.py]: @iamareebjamal - This init will override the default init created by dataclass.

**Comment by iamareebjamal** [comment]
We cannot change the behaviour of existing classes just to use dataclasses

It was discussed in previous dataclass PRs as well. Do what @mrsaicharan1 did. Learn to copy code. If it cannot be done using dataclasses, we obviousy cannot use dataclasses at the expense of breaking existing functionality. That's the Number 1 rule of refactoring - don't break existing functionality. Now you can understand what is to be done. Should you revert or should you not - whatever doesn't break existing functionality

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4820342&bid=16025263)

File: app/models/user_token_blacklist.py (hunk 2 above)
Comment (niranjan94) [inline app/models/user_token_blacklist.py]: Codacy Issue found: [invalid syntax](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4820342)

File: app/models/user_token_blacklist.py [hunk 4]
```diff
@@ -1,23 +1,18 @@
...
+    created_at: datetime = db.Column(db.DateTime(timezone=True), default=sql_func.now(), nullable=False)
+    blacklisted_at: datetime = db.Column(db.DateTime(timezone=True), default=sql_func.now(), nullable=False)
+    user_id: int = db.Column(db.Integer, db.ForeignKey('users.id', ondelete='CASCADE'), nullable=False)
+    user: int = db.relationship("User", backref="token_blacklist_times", foreign_keys=[user_id])
```
Comment (iamareebjamal) [inline app/models/user_token_blacklist.py]: This is not int

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4820342&bid=16025263)

File: app/models/user_token_blacklist.py [hunk 5]
```diff
@@ -1,23 +1,18 @@
...
 class UserTokenBlackListTime(db.Model):
     """user token blacklist time model class"""
     __tablename__ = 'user_token_blacklist_time'
     __table_args__ = (db.UniqueConstraint('user_id', name='user_blacklist_time_uc'),)
 
-    id = db.Column(db.Integer, primary_key=True)
-    created_at = db.Column(db.DateTime(timezone=True), default=sql_func.now(), nullable=False)
-    blacklisted_at = db.Column(db.DateTime(timezone=True), default=sql_func.now(), nullable=False)
-    user_id = db.Column(db.Integer, db.ForeignKey('users.id', ondelete='CASCADE'), nullable=False)
+    id: int = db.Column(db.Integer, primary_key=True)
```
Comment (niranjan94) [inline app/models/user_token_blacklist.py]: Codacy Issue found: [invalid syntax](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4820342)

**Comment by codedsun** [comment]
@iamareebjamal - Done the change!

**Pull request merged**